### DISCUSSION TITLE
Add tab_id and step_id as row-level parameter

### DISCUSSION THREAD

**Discussion by DSMejantel** [sqlpage/SQLPage#184] [Ideas]
Hello,
Just an idea. Is it possible to add id parameters to tab component (for step component too) ?
So it will be easy to jump from a tab to another tab with variables or numerical calculation...

**Comment by lovasoa** [comment]
@lorefnon is working on adding IDs to all components here: https://github.com/lovasoa/SQLpage/pull/182

However, I'm not sure how this would help in your case ? What prevents you from creating dynamic links for switching between tabs today ?

**Reply by DSMejantel** [reply]
Hi,
so we can have shorter link, but not so easier to understanding.
Il was an idea for a New "bee" project : to jump from a hive to the next hive.
Maybe in can do what in nées with variables.

**Reply by lorefnon** [reply]
Yeah, I am not sure I fully understand the use case either, but under the hood tabs are just links so we can easily propagate information already. 

The examples in the site often use static data for convenience/succinctness but we can construct the target url dynamically, and add query parameters from real data. For example:

```sql
select 
    'tab' as component;
select 
    'Sales performance' as title,
    $cur_tab_id = 'sales' as active,
    '?start_date=' || $start_date 
        || '&end_date=' || $end_date 
        || '&cur_tab_id=sales'
        || '&department_id=' || $dept_id as link;
select 
    'OKR progress' as title,
    $cur_tab_id = 'okr' as active,
    '?start_date=' || $start_date 
        || '&end_date=' || $end_date 
        || '&cur_tab_id=okr'
        || '&department_id=' || $dept_id as link;
```

And now in the linked page, we can just these variables (like `$start_date`, `$end_date`, `$department_id` etc.) as variables we can use in SQL queries.

The convenience of passing some adhoc query parameters  and being able to use them in SQL as parameters is really nice and was one of the primary reasons I started trying out sqlpage.

My PR will primarily enable targeting sections within a single page, but that is useful only primarily if the pages have a bunch of charts and tables (larger dashboards or reports).

**Comment by DSMejantel** [comment]
Thanks, 
It's right. We can work around my problem and do like you explain.

**Comment by DSMejantel** [comment]
My question was stupid.
We can do without id like this : 

```sql
--Onglets
SET tab=coalesce($tab,'1');
select 'tab' as component,
TRUE as center;
select  'Type de Ruche'  as title, 'home' as icon, 1  as active, 'parametres.sql?tab=1' as link, CASE WHEN $tab='1' THEN 'orange' ELSE 'green' END as color;
select  'Couleurs' as title, 'palette' as icon, 0 as active, 'parametres.sql?tab=2' as link, CASE WHEN $tab='2' THEN 'orange' ELSE 'green' END as color;
select  'Interventions' as title, 'tool' as icon, 1 as active, 'parametres.sql?tab=3' as link, CASE WHEN $tab='3' THEN 'orange' ELSE 'green' END as color;
select  'Variété de Miels' as title, 'flower' as icon, 1 as active, 'parametres.sql?**tab=4**' as link, CASE WHEN $tab='4' THEN 'orange' ELSE 'green' END as color;
```

![image](https://github.com/lovasoa/SQLpage/assets/149260856/73a832a3-842e-4429-9181-57164950e0e2)
